### PR TITLE
test: fix incorrect regexp pattern during migrating test

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Problem Summary:

https://github.com/pingcap/tidb/pull/30073#discussion_r755295895

### What is changed and how it works?

1. Add `^..$` to every pattern in require.Regexp.
2. Remove the beginning `^.*` or replace the beginning `^^` with `^`.
3. Remove the ending `.*$` or replace the ending `$$` with `$`.
4. Replace some `require.Regexp` to `require.Contains`.

### Check List

Tests 

- [x] Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: dumpling/export/prepare_test.go [hunk 1]
```diff
@@ -287,7 +287,8 @@ func TestConfigValidation(t *testing.T) {
 	conf.FileType = FileFormatSQLTextString
 	err := adjustFileFormat(conf)
 	require.Error(t, err)
-	require.Regexp(t, ".*please unset --filetype or set it to 'csv'.*", err.Error())
+	require.Regexp(t, "please unset --filetype or set it to 'csv'", err.Error())
```
Comment (gozssky) [inline dumpling/export/prepare_test.go:290]:
```suggestion
```

File: executor/show_stats_test.go [hunk 2]
```diff
@@ -258,7 +258,7 @@ func TestShowStatusSnapshot(t *testing.T) {
 	tk.MustQuery("show table status;").Check(testkit.Rows())
 	tk.MustExec("set @@tidb_snapshot = '" + snapshotTime.Format("2006-01-02 15:04:05.999999") + "'")
 	result := tk.MustQuery("show table status;")
-	require.Regexp(t, "t", result.Rows()[0][0])
+	require.Regexp(t, "^t$", result.Rows()[0][0])
```
Comment (gozssky) [inline executor/show_stats_test.go:261]:
```suggestion
	require.Equal(t, "t", result.Rows()[0][0])
```

File: br/pkg/metautil/metafile_test.go [hunk 3]
```diff
@@ -87,7 +86,7 @@ func TestWalkMetaFileInvalid(t *testing.T) {
 	collect := func(m *backuppb.MetaFile) { panic("unreachable") }
 	err := walkLeafMetaFile(ctx, mockStorage, root, &cipher, collect)
 
-	require.Regexp(t, regexp.MustCompile(".*ErrInvalidMetaFile.*"), err)
+	require.Contains(t, err.Error(), "ErrInvalidMetaFile")
```
Comment (kennytm) [inline br/pkg/metautil/metafile_test.go:89]:
```suggestion
	require.Error(t, err)
	require.Contains(t, err.Error(), "ErrInvalidMetaFile")
```

File: ddl/failtest/fail_db_serial_test.go [hunk 4]
```diff
@@ -568,6 +568,6 @@ func TestPartitionAddPanic(t *testing.T) {
 	_, err := tk.Exec(`alter table t add partition (partition p1 values less than (20));`)
 	require.Error(t, err)
 	result := tk.MustQuery("show create table t").Rows()[0][1]
-	require.Regexp(t, `(?s).*PARTITION .p0. VALUES LESS THAN \(10\).*`, result)
+	require.Regexp(t, `^(?s).*PARTITION .p0. VALUES LESS THAN \(10\)`, result)
 	require.NotRegexp(t, `(?s).*PARTITION .p0. VALUES LESS THAN \(20\).*`, result)
```
Comment (kennytm) [inline ddl/failtest/fail_db_serial_test.go:572]: change this NotRegexp too.

File: ddl/failtest/fail_db_serial_test.go [hunk 5]
```diff
@@ -568,6 +568,6 @@ func TestPartitionAddPanic(t *testing.T) {
 	_, err := tk.Exec(`alter table t add partition (partition p1 values less than (20));`)
 	require.Error(t, err)
 	result := tk.MustQuery("show create table t").Rows()[0][1]
-	require.Regexp(t, `(?s).*PARTITION .p0. VALUES LESS THAN \(10\).*`, result)
+	require.Regexp(t, `^(?s).*PARTITION .p0. VALUES LESS THAN \(10\)`, result)
```
Comment (kennytm) [inline ddl/failtest/fail_db_serial_test.go:571]:
```suggestion
	require.Regexp(t, `PARTITION .p0. VALUES LESS THAN \(10\)`, result)
```

File: expression/builtin_json_test.go [hunk 6]
```diff
@@ -119,7 +119,7 @@ func TestJSONUnquote(t *testing.T) {
 			require.Equal(t, tt.Result, d.GetString())
 			require.NoError(t, err)
 		} else {
-			require.Regexp(t, ".*The document root must not be followed by other values.*", err.Error())
+			require.Contains(t, err.Error(), "The document root must not be followed by other values")
```
Comment (kennytm) [inline expression/builtin_json_test.go:122]:
```suggestion
			require.Error(t, err)
			require.Contains(t, err.Error(), "The document root must not be followed by other values")
```

File: planner/core/integration_partition_test.go [hunk 7]
```diff
@@ -79,10 +79,10 @@ func TestListColVariousTypes(t *testing.T) {
 	tk.MustExec(`create table tstring (a varchar(32)) partition by list columns(a) (partition p0 values in ('a', 'b'), partition p1 values in ('c', 'd'))`)
 
 	err := tk.ExecToErr(`create table tdouble (a double) partition by list columns(a) (partition p0 values in (0, 1), partition p1 values in (2, 3))`)
-	require.Regexp(t, ".*not allowed.*", err)
+	require.Contains(t, err.Error(), "not allowed")
```
Comment (kennytm) [inline planner/core/integration_partition_test.go:82]:
```suggestion
	require.Error(t, err)
	require.Contains(t, err.Error(), "not allowed")
```

(and the rest of this file)

File: server/tidb_test.go [hunk 8]
```diff
@@ -1243,7 +1243,7 @@ func TestGracefulShutdown(t *testing.T) {
 
 	// nolint: bodyclose
 	_, err = cli.fetchStatus("/status") // status is gone
-	require.Regexp(t, ".*connect: connection refused", err.Error())
+	require.Regexp(t, "connect: connection refused$", err.Error())
```
Comment (kennytm) [inline server/tidb_test.go:1246]:
```suggestion
	require.Error(t, err)
	require.Regexp(t, "connect: connection refused$", err.Error())
```

File: sessionctx/variable/varsutil_test.go [hunk 9]
```diff
@@ -467,24 +467,24 @@ func TestVarsutil(t *testing.T) {
...
+	require.Regexp(t, "Incorrect argument type to variable 'tidb_stmt_summary_max_stmt_count'$", err)
 
 	err = SetSessionSystemVar(v, TiDBStmtSummaryMaxSQLLength, "10")
 	require.NoError(t, err)
 	val, err = GetSessionOrGlobalSystemVar(v, TiDBStmtSummaryMaxSQLLength)
 	require.NoError(t, err)
 	require.Equal(t, "10", val)
 	err = SetSessionSystemVar(v, TiDBStmtSummaryMaxSQLLength, "a")
-	require.Regexp(t, ".*Incorrect argument type to variable 'tidb_stmt_summary_max_sql_length'", err.Error())
+	require.Regexp(t, "Incorrect argument type to variable 'tidb_stmt_summary_max_sql_length'$", err.Error())
```
Comment (kennytm) [inline sessionctx/variable/varsutil_test.go:478]:
```suggestion
	require.Error(t, err)
	require.Regexp(t, "Incorrect argument type to variable 'tidb_stmt_summary_max_sql_length'$", err.Error())
```

File: types/datum_test.go [hunk 10]
```diff
@@ -201,7 +201,7 @@ func TestToFloat64(t *testing.T) {
 		if testCase.errMsg == "" {
 			require.NoError(t, err)
 		} else {
-			require.Regexp(t, testCase.errMsg, err)
+			require.Contains(t, err.Error(), testCase.errMsg)
```
Comment (kennytm) [inline types/datum_test.go:204]:
```suggestion
			require.Error(t, err)
			require.Contains(t, err.Error(), testCase.errMsg)
```

**Review by kennytm — COMMENTED** [review]
rest LGTM

**Review by kennytm — APPROVED** [review]

File: sessionctx/variable/varsutil_test.go [hunk 11]
```diff
@@ -467,24 +467,29 @@ func TestVarsutil(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "10", val)
 	err = SetSessionSystemVar(v, TiDBStmtSummaryMaxStmtCount, "a")
-	require.Regexp(t, ".*Incorrect argument type to variable 'tidb_stmt_summary_max_stmt_count'", err)
+	require.Error(t, err)
```
Comment (tisonkun) [inline sessionctx/variable/varsutil_test.go:470]: /cc @pingcap/tidb-configuration-reviewer

**Review by tisonkun — APPROVED** [review]
LGTM. But unfortunately any changes on `sessionctx/variable` requires someone from @pingcap/tidb-configuration-reviewer 's approval.

File: sessionctx/variable/varsutil_test.go (hunk 11 above)
Comment (gozssky) [reply sessionctx/variable/varsutil_test.go:470]: @morgo Could you please take a look? There are only test changes.

**Review by morgo — APPROVED** [review]

File: sessionctx/variable/varsutil_test.go (hunk 11 above)
Comment (morgo) [reply sessionctx/variable/varsutil_test.go:470]: Sorry for the delay. LGTM, but you now have a conflict.

Comment (gozssky) [reply sessionctx/variable/varsutil_test.go:470]: Fixed.

**Pull request merged**